CodenameOne/src/com/codename1/io/JSONParser.java

```diff
@@ -632,4 +634,270 @@ public void keyValue(String key, String value) {
+                } else {
+                    throw new Exception("Cast error in converting map to json");
+                }
+                json.append(value);
+            } else if (entry.getValue() instanceof Boolean) {
+                String value = ((Boolean) entry.getValue()).toString();
+                json.append(value);
+            } else if (entry.getValue() instanceof List) {
+                List myList = (List) (entry.getValue());
+                StringBuilder value = new StringBuilder();
+                value.append("[");
+                if (!myList.isEmpty() && (myList.get(0) instanceof String)) {
+                    for (int i = 0; i < myList.size(); i++) {
+                        String item = (String) (myList.get(i));
+                        value.append("\"").append((String) item).append("\"");
+                        if (i < myList.size() - 1) {
+                            value.append(",");
+                        }
+                    }
+                } else if (!myList.isEmpty() && myList.get(0) instanceof Number) {
+                    for (int i = 0; i < myList.size(); i++) {
+                        String item;
+                        if (myList.get(i) instanceof Byte) {
+                            item = ((Byte) myList.get(i)).toString();
+                        } else if (myList.get(i) instanceof Double) {
+                            item = ((Double) myList.get(i)).toString();
+                        } else if (myList.get(i) instanceof Float) {
+                            item = ((Float) myList.get(i)).toString();
+                        } else if (myList.get(i) instanceof Integer) {
+                            item = ((Integer) myList.get(i)).toString();
+                        } else if (myList.get(i) instanceof Long) {
+                            item = ((Long) myList.get(i)).toString();
+                        } else {
+                            throw new Exception("Cast error in converting map to json");
+                        }
+                        value.append((String) item);
+                        if (i < myList.size() - 1) {
+                            value.append(",");
+                        }
+                    }
+                } else if (!myList.isEmpty() && myList.get(0) instanceof Boolean) {
+                    for (int i = 0; i < myList.size(); i++) {
+                        Boolean item = (Boolean) (myList.get(i));
+                        value.append((Boolean) item);
+                        if (i < myList.size() - 1) {
+                            value.append(",");
+                        }
+                    }
+                } else if (!myList.isEmpty() && myList.get(0) instanceof Map) {
+                    for (int i = 0; i < myList.size(); i++) {
+                        String item = mapToJson((Map) (myList.get(i)));
+                        List<String> lines = StringUtil.tokenize(item, '\n');
```

Comment:
Inefficient. Use a helper method like `appendTo(StringBuilder result, Map<String, ?>, int depth)`.

And another one for lists.